### PR TITLE
MRG: Test SCI correctly checks file type at start of function

### PR DESCRIPTION
**Pull request by rob-luke**
#### Reference issue
General fNIRS tweaks #7057 

#### What does this implement/fix?
The previous check for the signal type was never reached as previous functions would fail. Moved check forward and added tests.

#### Additional information
Caught this one because I was going through the code coverage trying to hit 100%

### CODE REVIEW

File: mne/preprocessing/tests/test_scalp_coupling_index.py [hunk 1]
```diff
@@ -58,3 +61,7 @@ def test_scalp_coupling_index(fname, fmt, tmpdir):
     assert_allclose(sci[0:6], [1, 1, 1, 1, -1, -1], atol=0.01)
     assert np.abs(sci[6]) < 0.5
     assert np.abs(sci[7]) < 0.5
+
+    # Ensure function errors if wrong type is passed in
+    raw = beer_lambert_law(raw)
+    pytest.raises(RuntimeError, scalp_coupling_index, raw)
```
Comment (larsoner) [inline mne/preprocessing/tests/test_scalp_coupling_index.py]: same

File: mne/preprocessing/tests/test_scalp_coupling_index.py [hunk 2]
```diff
@@ -30,7 +31,9 @@
 def test_scalp_coupling_index(fname, fmt, tmpdir):
     """Test converting NIRX files."""
     assert fmt in ('nirx', 'fif')
-    raw = read_raw_nirx(fname)
+    raw = read_raw_nirx(fname).load_data()
+    pytest.raises(RuntimeError, scalp_coupling_index, raw)
```
[the remark this answers is not in the document]
Comment (rob-luke) [reply mne/preprocessing/tests/test_scalp_coupling_index.py]: Thanks @larsoner, appreciate the help.

**Comment by rob-luke** [comment]
@larsoner @agramfort could you please review this. Thanks

**Comment by agramfort** [comment]
thx @rob-luke

**Pull request merged**